src/main/java/eu/bbmri_eric/negotiator/settings/UIConfigServiceImpl.java

```diff
@@ -0,0 +1,84 @@
+package eu.bbmri_eric.negotiator.settings;
+
+import eu.bbmri_eric.negotiator.common.exceptions.WrongRequestException;
+import jakarta.validation.ConstraintViolation;
+import jakarta.validation.Validation;
+import jakarta.validation.Validator;
+import jakarta.validation.ValidatorFactory;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import org.springframework.stereotype.Service;
+
+@Service(value = "DefaultUIConfigService")
+public class UIConfigServiceImpl implements UIConfigService {
+
+  private final UIParameterRepository uiParameterRepository;
+
+  public UIConfigServiceImpl(UIParameterRepository uiParameterRepository) {
+    this.uiParameterRepository = uiParameterRepository;
+  }
+
+  @Override
+  public Map<String, Map<String, Object>> getAllParameters() {
+    List<UIParameter> parameters = uiParameterRepository.findAll();
+    Map<String, Map<String, Object>> settings = new HashMap<>();
+    parameters.forEach(
+        uiParameter -> {
+          switch (uiParameter.getType()) {
+            case INT:
+              settings
+                  .computeIfAbsent(uiParameter.getCategory(), k -> new HashMap<>())
+                  .put(uiParameter.getName(), Integer.valueOf(uiParameter.getValue()));
+              break;
+            case BOOL:
+              settings
+                  .computeIfAbsent(uiParameter.getCategory(), k -> new HashMap<>())
+                  .put(uiParameter.getName(), Boolean.valueOf(uiParameter.getValue()));
+              break;
+            default:
+              settings
+                  .computeIfAbsent(uiParameter.getCategory(), k -> new HashMap<>())
+                  .put(uiParameter.getName(), uiParameter.getValue());
+              break;
+          }
```

Comment:
I think this could be handled differently. Generally switch/if statements can get very long. Can't we put this logic into a private class in the Entity maybe

Reply:
I have added a getTypedValue to the UIParameter entity. It should shift the responsibility away from the service to the entity, which is the right responsible